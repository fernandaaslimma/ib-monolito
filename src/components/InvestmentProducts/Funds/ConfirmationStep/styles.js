import styled from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";
import { gray200, gray300 } from "../../../../styles/settings";
import { CheckboxSpan } from "../../../common/Checkbox/styles";

export const MainContainer = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const ResumeInfoWrapper = styled.div`
  margin: ${rem(24)} ${rem(16)} 0 ${rem(16)};
  width: ${rem(600)};
  padding-left: ${rem(16)};
  padding-right: ${rem(16)};

  @media (max-width: ${rem(600)}) {
    width: 100%;
  }
`;

export const AlertWrapper = styled.div`
  margin: ${rem(0)} ${rem(16)} 0 ${rem(16)};
  width: ${rem(600)};
  padding-left: ${rem(16)};
  padding-right: ${rem(16)};

  @media (max-width: ${rem(600)}) {
    width: 100%;
  }
`;

export const ResumeItemTitle = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(16)};
  line-height: 125%;
  color: ${gray300};

  margin-bottom: ${rem(8)};
  margin-right: ${rem(8)};
`;

export const ResumeItemTitleBold = styled.span`
  font-family: Lato;
  font-weight: 900;
  font-size: ${remFontSize(17.5)};
  font-scretch: condensed;
  line-height: ${rem(20)};
  color: ${gray300};

  margin-bottom: ${rem(8)};
  margin-right: ${rem(8)};
`;

export const ResumeItemValue = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(17)};
  color: ${gray200};
  margin: 0;
`;

export const SubItens = styled.div`
  display: flex;
  flex-direction: column;
  align-items: start;
  gap: ${rem(16)};
  margin-bottom: ${rem(24)};
`;

export const ResumeAmmount = styled.div`
  display: flex;
  align-items: center;
`;
export const Currency = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(12)};
  line-height: ${rem(14)};
  letter-spacing: 0.05em;
  margin-right: ${rem(3)};

  color: ${gray300};
`;
export const ResumeAmmountValue = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(24)};
  line-height: ${rem(29)};
  letter-spacing: 0.05em;
  color: ${gray300};
`;

export const Item = styled.div`
  margin-bottom: ${rem(24)};

  :last-child {
    margin-bottom: ${rem(32)};
  }
`;

export const ItemRemunerationInfo = styled.div`
  margin-bottom: ${rem(10)};
`;

export const ContractAdhesionFunds = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(16)};
  line-height: 125%;

  color: ${gray300};

  margin: ${rem(24)} 0 ${rem(16)} 0;
`;

export const ReadTerm = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(14)};
  line-height: 160%;
  letter-spacing: ${rem(0.452308)};
  text-decoration-line: underline;

  color: #3976cf;
  cursor: pointer;
`;

export const CheckboxContent = styled.div`
  margin: 0;
`;

export const CheckboxText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(17)};

  color: ${gray200};
`;

export const CheckboxTextBold = styled(CheckboxText)`
  font-weight: bold;
  color: ${gray300};
`;

export const CheckboxContainer = styled.span`
  ${CheckboxSpan} {
    position: relative;
    top: ${rem(-3)};
  }

  margin: ${rem(6)};
`;

export const InfoWrapper = styled.div`
  margin: ${rem(24)} ${rem(16)} ${rem(32)} ${rem(16)};
  display: flex;
  flex-direction: column;
`;

export const InfoRowWrapper = styled.div`
  display: flex;
  flex-direction: row;
`;

export const UnsuitableFundText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: 150%;
  align-items: center;
  letter-spacing: ${rem(0.45)};

  color: ${gray200};
`;

export const UnsuitableFundTextBold = styled(UnsuitableFundText)`
  font-weight: bold;

  color: ${gray300};
`;

export const InfoModalWrapper = styled.div`
  margin: ${rem(24)} ${rem(16)} ${rem(24)} ${rem(16)};
`;

export const InfoContent = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(15)};
  line-height: 150%;
  color: ${gray200};
  margin-bottom: ${rem(24)};

  b {
    font-weight: 600;
    color: ${gray300};
  }
`;

export const InfoContainer = styled.div`
  display: flex;
  justify-content: center;
  font-family: Lato;
  font-style: normal;
  font-weight: 400;
  font-size: ${remFontSize(14)};
  line-height: ${rem(21)};
  color: ${gray200};
  margin: ${rem(13)} ${rem(36)} ${rem(0)} ${rem(36)};
  min-height: ${rem(116)};
  align-items: center;

  @media (max-width: ${rem(600)}) {
    margin: ${rem(13)} ${rem(16)} ${rem(0)} ${rem(16)};
  }
`;

export const ButtonWrapper = styled.div`
  text-align: center;
  overflow: auto;
  margin: ${rem(6)} ${rem(8)} ${rem(22)} ${rem(8)};
`;

export const ButtonsWrapper = styled(ButtonWrapper)`
  display: flex;
`;

export const InfoContentBold = styled(InfoContent)`
  font-weight: 600;
  margin: 0;
  color: ${gray300};
`;

export const Message = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(14)};
  line-height: 150%;
`;

export const MessageBold = styled(Message)`
  font-weight: bold;
`;

export const StepVisibilityWrapper = styled.div`
  padding-left: ${rem(16)};
  padding-right: ${rem(16)};
  margin-top: ${rem(16)};
  width: ${rem(600)};

  @media (max-width: ${rem(600)}) {
    width: 100%;
  }
`;

export const TextInfoContainer = styled.span`
`;