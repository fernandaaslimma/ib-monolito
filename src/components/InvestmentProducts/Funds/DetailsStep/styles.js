import styled from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";
import { gray200, gray300 } from "../../../../styles/settings";

export const DetailHeader = styled.div`
  margin: ${rem(0)} ${rem(16)} ${rem(32)} ${rem(16)};
  padding-top: ${rem(27)};
`;
export const FundTypes = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: ${rem(3)};
`;
export const InvestmentType = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  font-size: ${remFontSize(16)};
  line-height: 125%;

  letter-spacing: ${rem(0.452308)};

  color: ${gray200};
`;
export const SuitabilityFundType = styled.h3``;

export const FundName = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(20)};
  line-height: 125%;
  align-items: left;
  width: ${rem(221)};
  margin-bottom: ${rem(8)};

  letter-spacing: ${rem(0.452308)};

  color: ${gray300};
`;

export const FundDescription = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: 160%;

  padding-top: ${rem(8)};

  letter-spacing: ${rem(0.452308)};

  color: ${gray200};
`;

export const QualifiedInvestorInfo = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  line-height: ${rem(20)};
  text-align: left;
  font-size: ${remFontSize(16)};
  letter-spacing: ${rem(0.452308)};
  color: ${gray200};
  padding: 0 ${rem(8)} 0 0;
`;

export const QualifiedInvestorWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: ${rem(8)};
`;

export const Wrapper = styled.div``;

export const ClickbleIcon = styled.div`
  cursor: pointer;
`;

export const InfoContent = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: 150%;
  letter-spacing: ${rem(0.45)};
  color: ${gray200};
  margin-bottom: ${rem(24)};

  b {
    font-weight: 600;
    color: ${gray300};
  }
`;

export const InfoContentBold = styled(InfoContent)`
  font-weight: 600;
  margin: 0;
  color: ${gray300};
`;

export const InfoWrapper = styled.div`
  display: flex;
  flex-direction: column;

  margin: ${rem(24)} ${rem(16)} ${rem(32)} ${rem(16)};
`;

export const ButtonWrapper = styled.div`
  margin: ${rem(16)} ${rem(-8)} ${rem(-8)} ${rem(-8)};
`;

export const ButtonsWrapper = styled(ButtonWrapper)`
  display: flex;
`;

export const AsteriskMessage = styled.div`
  margin: ${rem(12)} ${rem(8)} ${rem(0)} ${rem(8)};
`;

export const AsteriskTitle = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(14)};
  line-height: 125%;
  color: ${gray200};
`;

export const Content = styled.div`
  margin-bottom: ${rem(24)};
`;

export const InfoContentDetais = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  color: ${gray200};
  display: flex;
  flex-direction: column;
  margin-bottom: ${rem(16)};
`;

export const SubTitle = styled.span`
  font-weight: 700;
  font-size: ${remFontSize(16)};
  line-height: ${rem(20)};
  letter-spacing: ${rem(0.45)};
  color: ${gray300};
  margin: 0;
`;

export const ContentWrapper = styled.span`
  font-size: ${remFontSize(14)};
  line-height: ${rem(22, 4)};
  letter-spacing: ${rem(0.45)};
`;

export const ContentWrapper2 = styled.span`
  font-size: ${remFontSize(16)};
  line-height: ${rem(20)};
  letter-spacing: ${rem(0.45)};
  margin-top: ${rem(8)};
`;

export const FirstContentAvailar = styled.span`
  font-size: ${remFontSize(14)};
  line-height: ${rem(17, 5)};
  letter-spacing: ${rem(0.45)};
  display: flex;
  justify-content: space-between;
`;

export const ContentAvailar = styled.span`
  font-size: ${remFontSize(14)};
  line-height: ${rem(17, 5)};
  letter-spacing: ${rem(0.45)};
  display: flex;
  justify-content: space-between;
  margin-top: ${rem(14)};
`;

export const InfoContentResidents = styled.div`
  font-family: Lato;
  font-weight: normal;
  font-style: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(22, 4)};
  letter-spacing: ${rem(0.45)};
  color: ${gray200};
`;
