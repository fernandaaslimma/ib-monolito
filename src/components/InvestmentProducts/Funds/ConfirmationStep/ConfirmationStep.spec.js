import React from "react";
import { shallow } from "enzyme";
import ConfirmationStep from "./ConfirmationStep";
import { Button, Icon } from "react-bocombbm-components";
import AnimatedBottonSheet from "../../../common/AnimatedBottomSheet";
import { ReadTerm } from "./styles";
import Checkbox from "../../../common/Checkbox/Checkbox";
import {
  TYPE_PRODUCT_FUNDS,
  TYPE_PRODUCT_FUNDS_UNSUITABLE
} from "../../../../utils/constants";
import HideableValue from "../../../common/HideableValue";

jest.mock("../../../../utils/openFile");

const API = {
  id: 1,
  name: "BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO",
  cnpj: "29.733.842/0001-34",
  description: null,
  manager: "Bahia Asset Management",
  administrator: null,
  minimumBalance: 20000,
  minimumTransaction: 10000,
  initialInvestment: 20000,
  riskProfile: "Moderate",
  riskProfileLabel: "Moderado",
  returns: {
    twelveMonths: null,
    year: null,
    thirtySixMonths: null,
    fortyEightMonths: null,
    sinceInception: null
  },
  class: "Hedge Funds",
  classLabel: "Multimercado",
  subscription: {
    type: "Corrido",
    conversionDays: 20,
    settlementDays: 1
  },
  redemption: {
    type: "Corrido",
    conversionDays: 1,
    settlementDays: 0
  },
  benchmark: "CDI",
  quotaDate: "2020-12-21",
  administrationFee: 0.2,
  performanceFee: 0.2,
  inceptionDate: null,
  remunerationDetails: {
    autonomousAgent: null,
    distributor: {
      description: "Remuneração recebida pelo distribuidor: ",
      helpDescription:
        "De acordo coma norma xpto ... essas são as remunerações aferidas pelo distribuidor em caso de aplicação nese fundo",
      remunerations: [
        {
          description: "Perncentual da taxa de Administração: ",
          value: "10% ao ano",
          helpValueField: "Detalhes sobre como é feito o calculo da remuneração"
        },
        {
          description: "Perncentual da taxa de Perfomance: ",
          value: "10% ao ano",
          helpValueField: "Detalhes sobre como é feito o calculo da remuneração"
        }
      ]
    }
  }
};

const thisProps = {
  stepForward: jest.fn(),
  currentStep: 4,
  stepBack: jest.fn()
};

const props = {
  serverTime: jest.fn()
};

const termsToSign = {
  id: 145,
  productId: 1,
  idempotencyKey: "2b92510a-40e6-4a1e-9857-4741c5852e62",
  investmentValue: 40000.0,
  status: "AWAITING_APPROVAL",
  terms: [
    {
      name: "Term of adhesion BOCOM BBM CORPORATE CREDIT 60 FIC FIM CP",
      type: TYPE_PRODUCT_FUNDS,
      url:
        "https://api.bocombbm.com.br/productterms/fundos/adesao_37322143000120.pdf",
      id: 48
    },
    {
      name: "Non Compliance Term - Fixed Income",
      type: TYPE_PRODUCT_FUNDS_UNSUITABLE,
      url:
        "https://api.bocombbm.com.br/noncomplianceterms/fundos/desenquadramento_rf_mod.pdf",
      id: 250
    }
  ]
};

const termAdhesionToSign = {
  id: 145,
  productId: 1,
  idempotencyKey: "2b92510a-40e6-4a1e-9857-4741c5852e62",
  investmentValue: 40000.0,
  status: "AWAITING_APPROVAL",
  terms: [
    {
      name: "Term of adhesion BOCOM BBM CORPORATE CREDIT 60 FIC FIM CP",
      type: TYPE_PRODUCT_FUNDS,
      url:
        "https://api.bocombbm.com.br/productterms/fundos/adesao_37322143000120.pdf",
      id: 48
    }
  ]
};

const termNonComplianceToSign = {
  id: 145,
  productId: 1,
  idempotencyKey: "2b92510a-40e6-4a1e-9857-4741c5852e62",
  investmentValue: 40000.0,
  status: "AWAITING_APPROVAL",
  terms: [
    {
      name: "Non Compliance Term - Fixed Income",
      type: TYPE_PRODUCT_FUNDS_UNSUITABLE,
      url:
        "https://api.bocombbm.com.br/noncomplianceterms/fundos/desenquadramento_rf_mod.pdf",
      id: 250
    }
  ]
};

const context = {
  props,
  state: {
    selectedFund: API,
    selectedAccount: { accountNumber: "123456", availableBalance: 20000 },
    hasTerms: termsToSign,
    signed: true,
    investmentValue: "24,00"
  },
  createOperationFunction: jest.fn(),
  approveTerm: jest.fn(),
  downloadFile: jest.fn(),
  isMobile: jest.fn(),
  goToLinkRoute: jest.fn(),
  checkFundIsInMoviment: jest.fn()
};

const context2 = {
  props,
  state: {
    selectedFund: API,
    selectedAccount: { accountNumber: "123456", availableBalance: 20000 },
    hasTerms: termAdhesionToSign,
    signed: true,
    investmentValue: "24,00"
  },
  createOperationFunction: jest.fn(),
  approveTerm: jest.fn()
};

const context3 = {
  props,
  state: {
    selectedFund: API,
    selectedAccount: { accountNumber: "123456", availableBalance: 20000 },
    hasTerms: termNonComplianceToSign,
    signed: true,
    investmentValue: "24,00"
  },
  createOperationFunction: jest.fn(),
  approveTerm: jest.fn()
};

describe("ConfirmationStep component", () => {
  let setState, useStateSpy, useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementation(f => f());
  };
  beforeEach(() => {
    setState = jest.fn();
    useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(init => [init, setState]);
    React.useContext = jest.fn(() => context);

    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();
  });

  it.skip("Should match snapshot", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should match snapshot with screen 3", () => {
    const newProps = { ...thisProps, currentStep: 3 };
    const component = shallow(<ConfirmationStep {...newProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should Change Value Visibility to false", () => {
    const component = shallow(
      <ConfirmationStep {...thisProps}>
        <HideableValue hide={true} />
      </ConfirmationStep>
    );

    component.find(Icon).simulate("click");

    expect(component).toMatchSnapshot();
  });

  it.skip("Should Change Value Visibility to true", () => {
    const component = shallow(
      <ConfirmationStep {...thisProps}>
        <HideableValue hide={false} />
      </ConfirmationStep>
    );

    component.find(Icon).simulate("click");

    expect(component).toMatchSnapshot();
  });

  it.skip("Should render component with one term to approve (adhesion)", () => {
    React.useContext = jest.fn(() => context2);

    const component = shallow(<ConfirmationStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should render component with one term to approve (non compliance)", () => {
    React.useContext = jest.fn(() => context3);

    const component = shallow(<ConfirmationStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should render modal", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(Button)
      .at(0)
      .simulate("click");
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should click outside modal", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(AnimatedBottonSheet)
      .at(0)
      .prop("onClickInBack")();
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should declare that has read term(s)", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(ReadTerm)
      .at(0)
      .simulate("click");
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should declare that has read term(s)", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(ReadTerm)
      .at(1)
      .simulate("click");
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should check Checkbox", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component.find(Checkbox).prop("onChange")({ target: { checked: true } });
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should go to checkFundIsInMoviment modal", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(Button)
      .at(1)
      .simulate("click");
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should go to approve term func", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(Button)
      .at(2)
      .simulate("click");
    expect(context.approveTerm).toHaveBeenCalledWith(thisProps.goToStep);
  });

  it.skip("Should click outside modal 2", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(AnimatedBottonSheet)
      .at(1)
      .prop("onClickInBack")();
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should close the modal", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(Button)
      .at(3)
      .simulate("click");
    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should close the modal and go to createOperationFunction func", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(Button)
      .at(4)
      .simulate("click");
    expect(setState).toHaveBeenCalled();
    expect(context.createOperationFunction).toHaveBeenCalledWith(
      thisProps.goToStep
    );
  });

  it.skip("Should click outside modal 2", () => {
    const component = shallow(<ConfirmationStep {...thisProps} />);
    component
      .find(AnimatedBottonSheet)
      .at(1)
      .prop("onClickInBack")();
    expect(setState).toHaveBeenCalled();
  });
});
