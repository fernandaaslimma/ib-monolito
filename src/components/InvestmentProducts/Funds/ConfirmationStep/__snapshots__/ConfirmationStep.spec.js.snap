// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationStep component Should Change Value Visibility to false 1`] = `
<styled.div>
  <styled.div>
    <styled.span
      id="ConfirmationStep"
    />
    <styled.div>
      <styled.span />
      <styled.div>
        <HideableValue
          currency="R$"
          currencyColor="#99B5C6"
          hide={false}
          pattern="******"
          styles="font-size: 0.875rem; color: \\"#2d4758\\"; font-family: Lato Bold; margin-right: 0.75rem"
          value={20000}
        />
        <styled.span
          onClick={[Function]}
        >
          <Icon
            color="#3976CF"
            cursorPointer={true}
            height={20}
            spacing={
              Object {
                "bottom": "none",
                "left": "none",
                "right": "none",
                "top": "none",
              }
            }
            type="HideView"
            width={20}
          />
        </styled.span>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      key="0"
    >
      <styled.div>
        <styled.div>
          R$
        </styled.div>
        <styled.div />
      </styled.div>
    </styled.div>
    <styled.div
      key="1"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3>
        BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
      </styled.h3>
    </styled.div>
    <styled.div
      key="2"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3>
        20undefined
      </styled.h3>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3 />
    </styled.div>
    <styled.div
      key="4"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3 />
    </styled.div>
    <styled.div
      key="5"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3>
        06/01/2010
      </styled.h3>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.span />
    <styled.div>
      <Button
        dataTest="Button"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "xxs",
            "top": "s",
          }
        }
        type="outline"
      />
      <Button
        dataTest="continueInvestButton"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
      />
    </styled.div>
  </styled.div>
  <AnimatedBottonSheet
    dataTest="alertModal"
    head={
      Object {
        "title": undefined,
      }
    }
    isOpen={false}
    onClickInBack={[Function]}
    velocity={0.3}
  >
    <styled.div>
      <styled.div>
        <styled.span>
          undefined BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
        </styled.span>
        <styled.span
          onClick={[Function]}
        />
        <styled.span>
          Non Compliance Term - Fixed Income
        </styled.span>
        <styled.span
          onClick={[Function]}
        />
      </styled.div>
      <styled.span>
        <Checkbox
          checked={false}
          dataTest="checkboxAcceptAdhesionTermFunds"
          label={
            <styled.div>
              <styled.span>
                <Styled(styled.span)>
                  undefined BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO, Non Compliance Term - Fixed Income
                </Styled(styled.span)>
              </styled.span>
            </styled.div>
          }
          onChange={[Function]}
        />
      </styled.span>
      <Button
        dataTest="buttonAcceptAdhesionTermFunds"
        disabled={true}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
      />
    </styled.div>
  </AnimatedBottonSheet>
  <AnimatedBottonSheet
    ellipsis={true}
    head={
      Object {
        "title": "BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO",
      }
    }
    isOpen={false}
    onClickInBack={[Function]}
    velocity={0.3}
    widthTagEllipsis={true}
  >
    <styled.div>
      <styled.div>
        <styled.span
          data-test="hasMovementInProgress"
        >
          <Styled(styled.span) />
        </styled.span>
        <Styled(styled.div)>
          <styled.div>
            <Button
              dataTest="noButton"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
          <styled.div
            data-test="BtnWrapper"
          >
            <Button
              dataTest="yesButton"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
            />
          </styled.div>
        </Styled(styled.div)>
      </styled.div>
    </styled.div>
  </AnimatedBottonSheet>
  <AnimatedBottonSheet
    head={
      Object {
        "close": true,
        "icon": "Close",
        "title": null,
      }
    }
    isOpen={false}
    onClickInBack={[Function]}
    velocity={0.3}
  >
    <styled.div />
    <styled.div>
      <Button
        dataTest="Button"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "xxs",
            "left": "xxs",
            "right": "xxs",
            "top": "xxs",
          }
        }
      />
    </styled.div>
  </AnimatedBottonSheet>
</styled.div>
`;

exports[`ConfirmationStep component Should Change Value Visibility to true 1`] = `
<styled.div>
  <styled.div>
    <styled.span
      id="ConfirmationStep"
    />
    <styled.div>
      <styled.span />
      <styled.div>
        <HideableValue
          currency="R$"
          currencyColor="#99B5C6"
          hide={false}
          pattern="******"
          styles="font-size: 0.875rem; color: \\"#2d4758\\"; font-family: Lato Bold; margin-right: 0.75rem"
          value={20000}
        />
        <styled.span
          onClick={[Function]}
        >
          <Icon
            color="#3976CF"
            cursorPointer={true}
            height={20}
            spacing={
              Object {
                "bottom": "none",
                "left": "none",
                "right": "none",
                "top": "none",
              }
            }
            type="HideView"
            width={20}
          />
        </styled.span>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      key="0"
    >
      <styled.div>
        <styled.div>
          R$
        </styled.div>
        <styled.div />
      </styled.div>
    </styled.div>
    <styled.div
      key="1"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3>
        BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
      </styled.h3>
    </styled.div>
    <styled.div
      key="2"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3>
        20undefined
      </styled.h3>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3 />
    </styled.div>
    <styled.div
      key="4"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3 />
    </styled.div>
    <styled.div
      key="5"
    >
      <styled.div>
        <styled.h2 />
      </styled.div>
      <styled.h3>
        06/01/2010
      </styled.h3>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.span />
    <styled.div>
      <Button
        dataTest="Button"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "xxs",
            "top": "s",
          }
        }
        type="outline"
      />
      <Button
        dataTest="continueInvestButton"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
      />
    </styled.div>
  </styled.div>
  <AnimatedBottonSheet
    dataTest="alertModal"
    head={
      Object {
        "title": undefined,
      }
    }
    isOpen={false}
    onClickInBack={[Function]}
    velocity={0.3}
  >
    <styled.div>
      <styled.div>
        <styled.span>
          undefined BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
        </styled.span>
        <styled.span
          onClick={[Function]}
        />
        <styled.span>
          Non Compliance Term - Fixed Income
        </styled.span>
        <styled.span
          onClick={[Function]}
        />
      </styled.div>
      <styled.span>
        <Checkbox
          checked={false}
          dataTest="checkboxAcceptAdhesionTermFunds"
          label={
            <styled.div>
              <styled.span>
                <Styled(styled.span)>
                  undefined BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO, Non Compliance Term - Fixed Income
                </Styled(styled.span)>
              </styled.span>
            </styled.div>
          }
          onChange={[Function]}
        />
      </styled.span>
      <Button
        dataTest="buttonAcceptAdhesionTermFunds"
        disabled={true}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
      />
    </styled.div>
  </AnimatedBottonSheet>
  <AnimatedBottonSheet
    ellipsis={true}
    head={
      Object {
        "title": "BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO",
      }
    }
    isOpen={false}
    onClickInBack={[Function]}
    velocity={0.3}
    widthTagEllipsis={true}
  >
    <styled.div>
      <styled.div>
        <styled.span
          data-test="hasMovementInProgress"
        >
          <Styled(styled.span) />
        </styled.span>
        <Styled(styled.div)>
          <styled.div>
            <Button
              dataTest="noButton"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
          <styled.div
            data-test="BtnWrapper"
          >
            <Button
              dataTest="yesButton"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
            />
          </styled.div>
        </Styled(styled.div)>
      </styled.div>
    </styled.div>
  </AnimatedBottonSheet>
  <AnimatedBottonSheet
    head={
      Object {
        "close": true,
        "icon": "Close",
        "title": null,
      }
    }
    isOpen={false}
    onClickInBack={[Function]}
    velocity={0.3}
  >
    <styled.div />
    <styled.div>
      <Button
        dataTest="Button"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "xxs",
            "left": "xxs",
            "right": "xxs",
            "top": "xxs",
          }
        }
      />
    </styled.div>
  </AnimatedBottonSheet>
</styled.div>
`;

exports[`ConfirmationStep component Should match snapshot 1`] = `
<ConfirmationStep
  currentStep={4}
  stepBack={[MockFunction]}
  stepForward={[MockFunction]}
/>
`;

exports[`ConfirmationStep component Should match snapshot with screen 3 1`] = `
<ConfirmationStep
  currentStep={3}
  stepBack={[MockFunction]}
  stepForward={[MockFunction]}
/>
`;

exports[`ConfirmationStep component Should render component with one term to approve (adhesion) 1`] = `
<ConfirmationStep
  currentStep={4}
  stepBack={[MockFunction]}
  stepForward={[MockFunction]}
/>
`;

exports[`ConfirmationStep component Should render component with one term to approve (non compliance) 1`] = `
<ConfirmationStep
  currentStep={4}
  stepBack={[MockFunction]}
  stepForward={[MockFunction]}
/>
`;
